From f29df20548640af5549ca4f4d4058a6029e668d1 Mon Sep 17 00:00:00 2001
From: Ermine Jose <quic_erminejo@quicinc.com>
Date: Thu, 16 Mar 2023 15:47:45 +0530
Subject: [PATCH] include: override package configuration

when two package configuration with same name exist,
override the openwrt configuration. For example
the default configuration for busybox is taken from
package/utils/busybox, this change helps to override
with openwrt-patches/package/utils/busybox when
make defconfig is issued.

Change-Id: I9dab36ff7ab76cf61c8b066e1b0b866939f625ee
Signed-off-by: Ermine Jose <quic_erminejo@quicinc.com>
---
 include/package-dumpinfo.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/package-dumpinfo.mk b/include/package-dumpinfo.mk
index 597452e4b930..08bc514bcbb9 100644
--- a/include/package-dumpinfo.mk
+++ b/include/package-dumpinfo.mk
@@ -13,7 +13,10 @@ $(if $(PKG_BUILD_DEPENDS),Build-Depends: $(PKG_BUILD_DEPENDS)
 
 endef
 
+FindPackage ?=$(strip $(shell find $(TOPDIR)/openwrt-patches -name $(1) 2>/dev/null))
+
 define Dumpinfo/Package
+$(eval -include $(wildcard $(call FindPackage, $(1))/$(1).mk))
 $(info $(SOURCE_INFO)Package: $(1)
 $(if $(MENU),Menu: $(MENU)
 )$(if $(SUBMENU),Submenu: $(SUBMENU)
@@ -45,6 +48,7 @@ $(if $(URL),$(URL)
 )$(MAINTAINER)
 @@
 $(if $(Package/$(1)/config),Config:
+$(Package/$(1)/overrideconfig)
 $(Package/$(1)/config)
 @@
 ))
-- 
2.34.1

